cvat/apps/dataset_manager/task.py

```diff
@@ -477,11 +477,11 @@ def _delete(self, data=None):
 
     def delete(self, data=None):
         deleted_data = self._delete(data)
-        handle_annotations_change(self.db_job, deleted_data, "delete")
-
         if not self._data_is_empty(deleted_data):
             self._set_updated_date()
 
+        handle_annotations_change(self.db_job, deleted_data, "delete")
```

Comment:
Also not related to the PR, but we should fix it in the future as it looks critical:

We should not call `handle_annotations_change` within a postgres transaction block

E.g:
patch_job_data -> create -> handle_annotations_change
patch_job_data -> update -> handle_annotations_change
patch_job_data -> delete -> handle_annotations_change

As patch_job_data is performed in atomic transaction and all time it keeps connection opened to the database and locks some rows in the database. 

At the same time `handle_annotations_change` interacts with another database (Clickhouse)

Reply:
Basically, that's what transactions are created for - to roll back the whole operation, if it fails at any point. Need to think more abut it.

Reply:
With such logic we may just put all endpoints into transaction. 
Obviously, it will not work.

Transactions only for data consistency inside the database. 
That should not be responsible for external services.

Reply:
> With such logic we may just put all endpoints into transaction.
> Obviously, it will not work.

Actually, it's [one of the widespread approaches](https://docs.djangoproject.com/en/5.0/topics/db/transactions/#tying-transactions-to-http-requests). The only concern here is the performance, but it just means we have to avoid locks by using lock-free approaches and that long-running operations should be factored out into async calls.